.github/workflows/release-subsquid-processor.yml

```diff
@@ -0,0 +1,51 @@
+name: "Release Subsquid Production Image"
+env:
+  SUBSQUID_IMAGE_RELEASE_TAGS: ${{ env.DOCKER_REGISTRY_NAME }}/subsquid-processor:latest,${{ env.DOCKER_REGISTRY_NAME }}/subsquid-processor:{{ github.event.pull_request.head.sha }},v0.0.1
+on:
+  push:
+    tags:
+      - 'subsquid-processor-v[0-9]+\.[0-9]+\.[0-9]+'  # e.g. subsquid-processor-v0.0.1
+
+jobs:
+  test-build-publish-subsquid:
+    name: Test Subquid
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
```

Comment:
Since this is a push, `${{ github.event.pull_request.head.sha }}` will be `""`, the ref field can be omitted entirely I think.